### ISSUE TITLE
Wrong URLs created for assets in a page with date specified in directory name

### ISSUE THREAD

**Issue by roblillack** [Keats/gutenberg#1943]
# Bug Report

## Environment

Zola version: 0.16.0 (macOS)

## Expected Behavior
Calling `get_url(path=X)` with `X` being the path to a page asset should correctly create a resolvable URL. This should be true even for pages that have a date specified as part of their directory name. For example, for page `2022-07-29-post/index.md` with a page asset `img.jpg`, calling `get_url(path="/2022-07-29-post/img.jpg")` should lead to the URL `https://example.com/post/img.jpg`.

## Current Behavior
Currently, for a page `2022-07-29-post/index.md` with a page asset `img.jpg`, calling `get_url(path="/2022-07-29-post/img.jpg")` leads to the unresolvable URL `https://example.com/2022-07-29-post/img.jpg`.

## Step to reproduce

1. In an empty directory, create a simple site:

   ```sh
   echo 'base_url="https://example.com"' > config.toml
   mkdir -p content/2022-07-29-post templates
   touch content/2022-07-29-post/img.jpg
   cat > templates/page.html <<EOF
   {% for asset in page.assets %}
   <a href="{{ get_url(path=asset) }}">click</a>
   {%- endfor %}
   EOF
   cat > content/2022-07-29-post/index.md <<EOF
   ---
   title: Test
   ---
   EOF
   ```

2. Run `zola build`
3. Check the contents of `public/post/index.html`. It should contain
   ```html
   <a href="https://example.com/post/img.jpg">click</a>
   ```
   but contains
   ```html
   <a href="https://example.com/2022-07-29-post/img.jpg">click</a>
   ```

**Comment by Infides** [comment]
This is a known issue and a duplicate of: https://github.com/getzola/zola/issues/788 and https://github.com/getzola/zola/issues/1035

**Comment by Infides** [comment]
I am sorry to bring this topic up again. The behavior is still not solved with 0.17.x
I tried to find the root problem and I have the following theory:
When a page is created, the `from_file()` method from `page.rs` will be called and in this method the `find_related_assets()` method is called to find all assets. This will generate a vector with assets, which contain the path to the assets, in the case when a date is in the folder name, the date will also be in this path.
In order to see this I created a PR here: https://github.com/getzola/zola/pull/2146
This PR prints the paths for each asset, in the case with the date in the folder it will look like this:
```
Entry path: /tmp/.tmpGZlN57/content/posts/2013-06-02_with-assets
Entry path: /tmp/.tmpGZlN57/content/posts/2013-06-02_with-assets/fail.png
Entry path: /tmp/.tmpGZlN57/content/posts/2013-06-02_with-assets/graph.jpg
Entry path: /tmp/.tmpGZlN57/content/posts/2013-06-02_with-assets/example.js
```
Am I on the right track here? What do you think @Keats?

**Comment by Keats** [comment]
That represents the path to the asset in the filesystem, so it's normal to have the date in the path
